pkg/logs/tail.go

```diff
@@ -0,0 +1,603 @@
+				// since this is the head of the slice.
+				if len(recv) == 0 {
+					time.AfterFunc(c.cfg.sortBuffer, func() {
+						tdCh <- bufferedLog{
+							reqID: req,
+							seq:   highSeq,
+						}
+					})
+				}
+
+				// Set the new log and receive info back to the
+				// logmap for this RequestID.
+				logmap[req] = reqLogs
+			}
+
+		case bufdLogs := <-tdCh: // A timer expired for a particular request.
+			reqID, seq := bufdLogs.reqID, bufdLogs.seq
+
+			// Get the logs for this RequestID and
+			// find the index of the sequence in our current logs.
+			reqLogs := logmap[reqID]
+			idx := findIdxBySeq(reqLogs.logs, seq)
+
+			// Split off the source of this timer, leave
+			// remaining logs to be printed later.
+			toPrint, remainingLogs := reqLogs.logs[:idx], reqLogs.logs[idx:]
+			reqLogs.logs = remainingLogs
+			c.printLogs(out, toPrint)
+
+			// Special case if we just printed the entire set of
+			// logs, we remove the keys from the maps and finish.
+			if len(remainingLogs) == 0 {
+				delete(logmap, reqID)
+				break
+			}
+
+			// Drop the front of the batchReqReceives map and start
+			// another timer for any remaining recorded sequences.
+			recv := reqLogs.receives[1:]
+			reqLogs.receives = recv
+
+			// If anything is left...
+			if len(recv) > 0 {
+				// We create a new timer, we subtract
+				// off time already served from the
+				// user defined sortBuffer.
+				time.AfterFunc(c.cfg.sortBuffer-time.Since(recv[0].when), func() {
+					tdCh <- bufferedLog{
+						reqID: reqID,
+						seq:   recv[0].highSeq,
+					}
+				})
+			}
+
+			// Set the new log and receive info back to the
+			// logmap for this RequestID.
+			logmap[reqID] = reqLogs
+
+		case <-c.doneCh:
+			os.Exit(0)
+		}
+	}
+}
+
+// printLogs is a simple printer for Log slices, only printing requested
+// streams.
+func (c *TailCommand) printLogs(out io.Writer, logs []Log) {
+	if len(logs) > 0 {
+		filtered := filterStream(c.cfg.stream, logs)
+
+		for _, l := range filtered {
+			text.Output(out, l.String())
+		}
+	}
+}
+
+// doReq runs the http.Request, returning a http.Response or error.
+func (c *TailCommand) doReq(req *http.Request) (*http.Response, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	req = req.WithContext(ctx)
+	go func() {
+		select {
+		case <-ctx.Done():
+		case <-c.dieCh:
+			cancel()
+		}
+	}()
+
+	resp, err := c.hClient.Do(req)
+	return resp, err
+}
+
+//
+// Log
+//
+
+// RequestStartFromRaw return a time.Time object representing the
+// RequestStart data.
+func (l *Log) RequestStartFromRaw() time.Time {
+	// RequestTime comes as unix time in microseconds. Convert to
+	// nanoseconds, then parse with stdlib.
+	nano := l.RequestStart * 1000
+	return time.Unix(0, nano)
+}
+
+// String is used to print a log for the tail output.
+func (l *Log) String() string {
+	// Trim the RequestID for nicer output, it might be a long UUID.
+	return fmt.Sprintf("| %6s | %8.8s | %s |",
```

Comment:
We should lose the left most pipe:

```suggestion
	return fmt.Sprintf("%6s | %8.8s | %s |",
```

Reply:
Removed leading and trailing separator.  That will make using `cut` easier if need be